src/Couchbase.Lite.Shared/API/Error/CouchbaseLiteErrorMessage.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Couchbase.Lite
+{
+    internal static class CouchbaseLiteErrorMessage
+    {
+        //Database error messages
+        internal const string CreateDBDirectoryFailed = "Unable to create database directory";
+        //Database - Copy
+        internal const string ResolveDefaultDirectoryFailed = "Failed to resolve a default directory! If you have overriden the IDefaultDirectoryResolver interface, please check it.  Otherwise please file a bug report.";
+        internal const string InvalidPath = "Path.Combine failed to return a non-null value!";
+        //Database - Close
+        internal const string CloseDBFailedReplications = "Cannot close the database. Please stop all of the replicators before closing the database.";
+        internal const string CloseDBFailedQueryListeners = "Cannot close the database. Please remove all of the query listeners before closing the database.";
+        //Database - Delete (Save with deletion =  true)
+        internal const string DeleteDBFailedReplications = "Cannot delete the database. Please stop all of the replicators before closing the database.";
+        internal const string DeleteDBFailedQueryListeners = "Cannot delete the database. Please remove all of the query listeners before closing the database.";
+        internal const string DeleteDocFailedNotSaved = "Cannot delete a document that has not yet been saved";
+        //Database - Purge
+        internal const string DocumentNotFound = "Document doesn't exist in the database.";
+        internal const string DocumentAnotherDatabase = "Cannot operate on a document from another database";
+        //Database - Save
+        internal const string BlobDifferentDatabase = "A document contains a blob that was saved to a different database; the save operation cannot complete.";
+        internal const string BlobContentNull = "No data available to write for install. Please ensure that all blobs in the document have non-null content.";
+        //Database - Resolve Conflict
+        internal const string ResolvedDocContainsNull = "Resolved document contains a null body";
+        internal const string ResolvedDocFailedLiteCore = "LiteCore failed resolving conflict.";
+        internal const string ResolvedDocWrongDb = "Resolved document db {0} is different from expected db {1}";
```

Comment:
{0}  = is this a placeholder or just string, pointing to the first argument printed after this line.

Reply:
It's a format placeholder.  The first and second arguments passed to `String.Format` on .NET will be used in the final message.

Reply:
is the compiler generate respective cross platform code placeholders for us.

Reply:
Yes, I think it should.